### PR TITLE
Add ability to specify model Revision with ":" syntax

### PR DESCRIPTION
**Pull request by deoxykev**
## Details

This allows you to specify the model revision (AKA branch) for a huggingface model by using the ":" convention. For example: `myusername/mymodelname:branchv1`

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/OpenRouterTeam/openrouter-runner/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I agree to license this contribution under the MIT LICENSE
- [x] I checked the [current PR](https://github.com/OpenRouterTeam/openrouter-runner/pulls) for duplication.

### CODE REVIEW

File: modal/runner/shared/download.py [hunk 1]
```diff
@@ -22,6 +22,13 @@ def download_models(all_models: List[str]):
 
     cache_path = get_model_path("__cache__")
     for model_name in all_models:
+        parts = model_name.split(":")
+
+        model_revision = None
```
Comment (louisgv) [inline modal/runner/shared/download.py:27]: This only downloads the model - can we also added it somehow into the protocol's input params?

**Comment by deoxykev** [comment]
What do you mean by protocol's input params? I looked around the codebase, but I'm not quite sure what you mean. Can you point out the relevant section?

**Comment by sambarnes** [comment]
> What do you mean by protocol's input params? I looked around the codebase, but I'm not quite sure what you mean. Can you point out the relevant section?

@deoxykev IIUC he's referring to a spot like this:
https://github.com/OpenRouterTeam/openrouter-runner/blob/main/modal/shared/protocol.py#L27

to make sure that its usages in the API take branch name into account somehow

**Comment by deoxykev** [comment]
@louisgv

Ok, now you are able to specify the revision in the API. I didn't modify `protocol.py`. Instead, the revision is just part of the model parameter. If no branch is specified, it loads main by default, so the original API calling conventions remain the same.

#### Example
- Specify models with revisions like this in `modal/engines/__init__.py`:
```python
vllm_7b_model_ids = [
    "TheBloke/law-LLM-GPTQ:gptq-4-32g-actorder_True",
    "TheBloke/law-LLM-GPTQ:gptq-4-16g-actorder_True",
    "TheBloke/law-LLM-GPTQ",
 ]
```

- Now they can be called by API using the optional ":revision" convention in the `model` parameter.

```bash
pnpm x scripts/test-simple.ts "TheBloke/law-LLM-GPTQ"
pnpm x scripts/test-simple.ts "TheBloke/law-LLM-GPTQ:gptq-4-32g-actorder_True"
pnpm x scripts/test-simple.ts "TheBloke/law-LLM-GPTQ:gptq-4-64g-actorder_True"
```

**Comment by louisgv** [comment]
Awesome 👍  - BTW the linter failed, but the overall logic lgtm

**Comment by deoxykev** [comment]
@louisgv Fixed linter issue, should be good to go now.

**Comment by louisgv** [comment]
Thanks @deoxykev - the linter is still complaining due to newline mismatching, but lgtm overall.

**Pull request merged**